app/src/main/java/app/grapheneos/camera/ui/ZoomableImageView.kt

```diff
@@ -123,10 +141,112 @@ class ZoomableImageView @JvmOverloads constructor(
         }
     }
 
+    private var lastClickTimestampMs : Long = 0L
+
+    private fun onClickEvent(event : MotionEvent) {
+        if ((event.eventTime - lastClickTimestampMs) <= DOUBLE_TAP_DELAY) {
+            singleClickHandler.removeCallbacks(singleClickRunnable)
+            onDoubleClick(event)
+        } else {
+            singleClickHandler.postDelayed(singleClickRunnable, DOUBLE_TAP_DELAY)
+        }
+        lastClickTimestampMs = event.eventTime
+    }
+
+    private fun onSingleClick() {
+        gActivity.toggleActionBarState()
+    }
+
+    private fun onDoubleClick(event: MotionEvent) {
+        // The user hasn't zoomed in
+        if (saveScale != 1f) {
+            scaleAnimator?.cancel()
+            scaleAnimator = ValueAnimator.ofFloat(saveScale, 1f)
+            scaleAnimator?.duration = SCALE_ANIMATION_DURATION
+            scaleAnimator?.addUpdateListener {
+                scaleImageTo(width / 2f, height / 2f, it.animatedValue as Float)
+            }
+
+            scaleAnimator?.addListener(object: AnimatorListener {
+
+                var isAnimationCancelled = false
+
+                override fun onAnimationStart(p0: Animator) {}
```

Comment:
Proper parameter names are missing, likely due to lack of Android SDK sources. Check whether they are downloaded in Android Studio SDK manager

Reply:
Have fixed in the changes that were pushed recently. Thanks!